examples/actix-web.rs

```diff
@@ -0,0 +1,73 @@
+use std::sync::Mutex;
+
+use actix_web::{web, App, HttpResponse, HttpServer, Responder, Result};
+use prometheus_client::encoding::text::{encode, Encode};
+use prometheus_client::metrics::counter::Counter;
+use prometheus_client::metrics::family::Family;
+use prometheus_client::registry::Registry;
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
+pub enum Method {
+    Get,
+    Post,
+}
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
+pub struct MethodLabels {
+    pub method: Method,
+}
+
+pub struct Metrics {
+    requests: Family<MethodLabels, Counter>,
+}
+
+impl Metrics {
+    pub fn inc_requests(&self, method: Method) {
+        self.requests.get_or_create(&MethodLabels { method }).inc();
+    }
+}
+
+pub struct AppState {
+    pub registry: Registry,
+}
+
+pub async fn metrics_handler(state: web::Data<Mutex<AppState>>) -> Result<HttpResponse> {
+    let state = state.lock().unwrap();
+    let mut buf = Vec::new();
+    encode(&mut buf, &state.registry)?;
+    let body = std::str::from_utf8(buf.as_slice()).unwrap().to_string();
+    Ok(HttpResponse::Ok().body(body))
```

Comment:
Would one not need to set the content type?

https://github.com/prometheus/client_rust/blob/6b7a3458b47a406d15f2ae0a678c611969c4645d/examples/tide.rs#L37

Reply:
Actually no, my experience shows that it is not required. Everything works well without this detail.

Reply:
According to the OpenMetrics specification it "MUST" be set. Am I missing something @oriontvv?

https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#overall-structure

Reply:
//CC @RichiH

Reply:
Okay, I'll add the header. (btw, we use at work without it and everything is fine)